### PR TITLE
Emulate reduced-motion in e2e tests

### PR DESCRIPTION
**Pull request by kevin940726**
## Description

Originally brought up in https://github.com/WordPress/gutenberg/pull/33317#discussion_r667744967. Emulate `prefers-reduced-motion: reduce` in e2e tests.

## How has this been tested?

E2E tests should pass

## Types of changes

New feature

## Checklist:
- [x] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/e2e-tests/config/setup-test-framework.js [hunk 1]
```diff
@@ -240,6 +240,9 @@ beforeAll( async () => {
 	await trashAllPosts( 'wp_block' );
 	await setupBrowser();
 	await activatePlugin( 'gutenberg-test-plugin-disables-the-css-animations' );
```
Comment (youknowriad) [inline packages/e2e-tests/config/setup-test-framework.js:242]: I think this means we can remove that plugin no?

File: .github/workflows/end2end-test.yml [hunk 2]
```diff
@@ -38,7 +38,7 @@ jobs:
             - name: Npm install and build
               run: |
                   npm ci
-                  FORCE_REDUCED_MOTION=true npm run build
```
Comment (youknowriad) [inline .github/workflows/end2end-test.yml:41]: should we remove the usage of `FORCE_REDUCED_MOTION` in the code.

**Review by youknowriad — COMMENTED** [review]
nice one 👍

File: packages/e2e-tests/config/setup-test-framework.js (hunk 1 above)
Comment (kevin940726) [reply packages/e2e-tests/config/setup-test-framework.js:242]: Yep, but I'd rather keep it for a while though. As mentioned in the original issue, I don't think we're quite there yet to disable this plugin entirely. Disabling it now would probably slow down the tests a little too.

File: .github/workflows/end2end-test.yml (hunk 2 above)
Comment (kevin940726) [reply .github/workflows/end2end-test.yml:41]: I'm not sure if it's used by anyone or anything else. But sure, I think we should remove it.

File: packages/compose/src/hooks/use-reduced-motion/index.js [hunk 3]
```diff
@@ -17,9 +17,8 @@ const IS_IE =
  *
  * @return {boolean} Reduced motion preference value.
  */
-const useReducedMotion =
-	process.env.FORCE_REDUCED_MOTION || IS_IE
-		? () => true
-		: () => useMediaQuery( '(prefers-reduced-motion: reduce)' );
+const useReducedMotion = IS_IE
```
Comment (youknowriad) [inline packages/compose/src/hooks/use-reduced-motion/index.js]: I think we can remove the is_ie check now :)

**Review by youknowriad — APPROVED** [review]
Nice one

**Pull request merged**